include/exadg/incompressible_navier_stokes/spatial_discretization/operator_coupled.cpp

```diff
@@ -329,6 +337,14 @@ OperatorCoupled<dim, Number>::evaluate_nonlinear_residual_steady(BlockVectorType
                                                                  BlockVectorType const & src,
                                                                  double const &          time) const
 {
+  // update implicitly coupled variable viscosity
+  if(this->param.viscous_term_is_nonlinear() &&
+     this->param.viscosity_model_data.treatment_of_nonlinear_viscosity ==
+       TreatmentOfNonlinearViscosity::Implicit)
+  {
+    this->update_viscosity(src.block(0));
+  }
```

Comment:
We have these lines of code more than once, so I wonder whether it could make sense to write the if-condition inside the function `update_viscosity()`? I think we should refactor this to ease maintainability.